pkgs/build-support/rust/build-rust-crate/configure-crate.nix

```diff
@@ -161,14 +165,24 @@ in ''
        EXTRA_BUILD_FLAGS="$EXTRA_BUILD_FLAGS $(tr '\n' ' ' < target/link.build)"
      fi
      noisily rustc --crate-name build_script_build $BUILD --crate-type bin ${rustcOpts} \
-       ${crateFeatures} --out-dir target/build/${crateName} --emit=dep-info,link \
+       ${mkRustcFeatureArgs crateFeatures} --out-dir target/build/${crateName} --emit=dep-info,link \
        -L dependency=target/buildDeps ${buildDeps} --cap-lints allow $EXTRA_BUILD_FLAGS --color ${colors}
 
      mkdir -p target/build/${crateName}.out
      export RUST_BACKTRACE=1
      BUILD_OUT_DIR="-L $OUT_DIR"
      mkdir -p $OUT_DIR
-     target/build/${crateName}/build_script_build > target/build/${crateName}.opt
+
+     (
+       # Features should be set as environment variable for build scripts:
+       # https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-build-scripts
+       for feature in ${envFeatures}; do
+         export CARGO_FEATURE_$feature=1
```

Comment:
are features allowed to contain whitespaces? Is this the sanest way to set them?

Reply:
Are there other possible values?

Reply:
> are features allowed to contain whitespaces?

Yes, they are, though I don’t believe anybody actually uses features with spaces in them. Still a worthwhile corner-case to handle.

Reply:
Value is hardcoded to 1 in cargo 

https://github.com/rust-lang/cargo/blob/c719a05a81339ec6626965871fb93dcc6be8c14c/src/cargo/core/compiler/custom_build.rs#L213-L215

Reply:
> Yes, they are, though I don’t believe anybody actually uses features with spaces in them. Still a worthwhile corner-case to handle.

I didn't treat any other cases, since Cargo does not do anything besides uppercasing and replacing dashes either?:

https://github.com/rust-lang/cargo/blob/ee417cbc2fea5c63bb32f91af53e64a3ccbb381e/src/cargo/core/compiler/mod.rs#L1098

Reply:
Interesting, Cargo will set the environment variable with a space in it. So I guess we'll have to do that too.

Reply:
Currently fails seems to fail in other places. Once a feature contains a space, it is also incorrectly passed to rustc and already fails when compiling the build script.

I guess supporting features with spaces would be better to handle in a separate PR.

Reply:
Yeah, I agree. Is anyone going to do that? ;)

Reply:
I have created an issue #90605 . I'll try to have a look at it one of the coming days.